### PR TITLE
Update contribution guide, with additional info on Go and Mongo versions

### PR DESCRIPTION
**Pull request by flimzy**
This updates CONTRIBUTING.md to provide additional information to new developers, about which versions of Go are supported, and how to get tests workging with supported versions of MongoDB.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -33,24 +33,33 @@ Building Juju
...
+module github.com/juju/juju
+
+go 1.14
+```
 
-[Snap](https://snapcraft.io/go) is the recommended way to install Go on Linux.
+### Official distribution
 
-### Other options
+Go can be [installed](https://golang.org/doc/install#install) Go from the official distribution.
```
Comment (markelog) [inline CONTRIBUTING.md:48]: Looks like the second `Go` is redundant?

**Comment by manadart** [comment]
We need to target this at 2.9 and fill out the patch description.

**Comment by manadart** [comment]
It looks the prior formatting included line-breaks at or around the 80 character mark, can we preserve this?

**Review by hpidcock — APPROVED** [review]
Going to approve this and follow up with updates.

**Comment by hpidcock** [comment]
$$merge$$

**Pull request merged**